submissions/gabrielks_brute_force_1.jl

```diff
@@ -0,0 +1,187 @@
+    return max_precipitation(parameter_tuple)
+end
+
+function max_precipitation(parameters::NamedTuple)
+
+    # define resolution. Use trunc=42, 63, 85, 127, ... for higher resolution, cubically slower
+    spectral_grid = SpectralGrid(trunc=31, nlayers=8)
+
+    # Define AquaPlanet ocean, for idealised sea surface temperatures
+    # but don't change land-sea mask = retain real ocean basins
+    ocean = AquaPlanet(spectral_grid,
+                temp_equator=parameters.temperature_equator,
+                temp_poles=parameters.temperature_pole)
+
+    land_temperature = ConstantLandTemperature(spectral_grid)
+    land = LandModel(spectral_grid; temperature=land_temperature)
+
+    initial_conditions = InitialConditions(
+        vordiv = ZonalWind(u₀=parameters.zonal_wind),
+        temp = JablonowskiTemperature(u₀=parameters.zonal_wind),
+        pres = PressureOnOrography(),
+        humid = ConstantRelativeHumidity())
+
+    orography = EarthOrography(spectral_grid, scale=parameters.orography_scale)
+
+    # construct model
+    model = PrimitiveWetModel(spectral_grid; ocean, land, initial_conditions, orography)
+
+    # Add rain gauge, locate in Pittsburgh PA
+    rain_gauge = RainGauge(spectral_grid, lond=-80, latd=40.45)
+    add!(model, rain_gauge)
+
+    # Initialize
+    simulation = initialize!(model, time=DateTime(2025, 7, 22))
+
+    # Add additional  mountain
+    H = parameters.mountain_height
+    λ₀, φ₀, σ = parameters.mountain_lon, parameters.mountain_lat, parameters.mountain_size  
+    set!(model, orography=(λ,φ) -> H*exp(-spherical_distance((λ,φ), (λ₀,φ₀), radius=360/2π)^2/2σ^2), add=true)
+
+    # land sea surface temperature anomalies
+    # 1. USA
+    set!(simulation, soil_temperature=
+        (λ, φ, k) -> (30 < φ < 50) && (240 < λ < 285) ? parameters.temperature_usa : 0, add=true)
+
+    # 2. Pennsylvania
+    A = parameters.temperature_pa
+    λ_az, φ_az, σ_az = -80, 40.45, 4    # location [˚], size [˚] of Azores
+    set!(simulation, soil_temperature=
+        (λ, φ, k) -> A*exp(-spherical_distance((λ,φ), (λ_az,φ_az), radius=360/2π)^2/2σ_az^2), add=true)
+
+    # Run simulation for 20 days
+    run!(simulation, period=Day(20))
+
+    # skip first 5 days, as is done in the RainMaker challenge
+    RainMaker.skip!(rain_gauge, Day(5))
+
+    # evaluate rain gauge
+    lsc = rain_gauge.accumulated_rain_large_scale
+    conv = rain_gauge.accumulated_rain_convection
+    total_precip = maximum(lsc) + maximum(conv)
+    return rain_gauge, total_precip
+end
+
+final_params = 
+[2.0163488388061523,
+ 5013.733625411987,
+ -0.017316662706434727,
+ -22.202972173690796,
+ -19.587678909301758,
+ 282.97233670949936,
+ 300.200936794281,
+ 2.133525013923645,
+ 4.704113602638245,
+ 34.923005402088165]
+
+rain_gauge, total_precip = max_precipitation(final_params)
+
+total_precip  # produces 212.7933 on RainMakerChallenge2025.jl's evaluator, 211.040 on this evaluator -- TODO investigate the discrepancy
```

Comment:
SpeedyWeather is not bit reproducible across different hardware as we use things like `muladd` that may or may not be compiled to a fused multiply-add depending on the chips instruction set